source/common/api/win32/os_sys_calls_impl.cc

```diff
@@ -73,23 +74,22 @@ LPFN_WSARECVMSG getFnPtrWSARecvMsg() {
   return recvmsg_fn_ptr;
 }
 
-using WSAMSGPtr = std::unique_ptr<WSAMSG>;
-
-WSAMSGPtr msghdrToWSAMSG(const msghdr* msg) {
+wsabufResult msghdrToWSAMSG(const msghdr* msg) {
   WSAMSGPtr wsa_msg(new WSAMSG);
 
   wsa_msg->name = reinterpret_cast<SOCKADDR*>(msg->msg_name);
   wsa_msg->namelen = msg->msg_namelen;
-  wsabufResult wsabuf = iovecToWSABUF(msg->msg_iov, msg->msg_iovlen);
-  wsa_msg->lpBuffers = wsabuf.wsabuf_.get();
-  wsa_msg->dwBufferCount = wsabuf.num_vec_;
+  auto buffer = iovecToWSABUF(msg->msg_iov, msg->msg_iovlen);
+  wsa_msg->lpBuffers = (buffer.size() > 0) ? &buffer[0] : NULL;
```

Comment:
`&buffer[0]` Is this an accepted pattern in C++? Seems we are relying on the internal layout of std::vector?

Reply:
Yeah it is guaranteed by the standard that this is valid C++ [ref](https://isocpp.org/wiki/faq/containers#vector-is-contiguous).
  
That being said probably using `data()` is a better version of this for C++1x so I will change it.

Reply:
The elt deference or data() can be used. https://en.cppreference.com/w/cpp/container/vector/data gives more insight and suggests we want to use data(), but it appears that until we moved to stdc++17 this old syntax was more appropriate.

Reply:
I swapped out to `data()` in 0b477d9efcbf75b5a193b70ef7e441a33b8afb85 because it shows better intent and is less confusing if you are not sure about the C++ standard